Library/ViewModels/DiscoveryPageViewModel.swift

```diff
@@ -305,25 +308,53 @@ public final class DiscoveryPageViewModel: DiscoveryPageViewModelType, Discovery
         AppEnvironment.current.koala.trackDiscoveryPullToRefresh()
       }
 
-    self.showEditorialHeader = Signal.combineLatest(currentUser, self.sortProperty.signal.skipNil())
-      .filter { _ in featureGoRewardlessIsEnabled() }
-      .map(second)
-      .filter { $0 == .magic }
-      .skipRepeats()
-      .map { _ in
-        DiscoveryEditorialCellValue(
+    // MARK: - Editorial Header
+
+    let filtersUpdated = self.sortProperty.signal.skipNil()
+      .takePairWhen(self.selectedFilterProperty.signal.skipNil().skipRepeats())
+
+    let editorialHeaderShouldShow = filtersUpdated
+      .filter { sort, _ in
+        sort == .magic
+      }
+      .map { args -> Bool in
+        let (sort, filterParams) = args
+
+        return sort == .magic && filterParams == DiscoveryViewModel.initialParams()
+      }
+
+    let updatedConfig = self.configUpdatedProperty.signal.skipNil()
```

Comment:
Would something like this help to ensure a config is read when this view appears? 🤔 

```swift
let updatedConfig = self.configUpdatedProperty.signal.skipNil()
let currentConfig = self.viewWillAppearProperty.signal.map { _ in AppEnvironment.current.config }
  .skipNil()
  .take(until: updatedConfig.ignoreValues())

let config = Signal.merge(
  updatedConfig,
  currentConfig
)
```

Reply:
I added debouncing, lmk what you think.

Reply:
If the aim is to ensure that this emits _after_ the `UITableView` has reloaded I just wonder if it wouldn't be better to have an input informing us that it's reloaded instead of arbitrarily debouncing this emission 🤔

Reply:
Oh so actually that's not the issue here - the issue is that there's sometimes a race condition between the config updating and the VC being initialized, whereby if the `ksr_configUpdated` notification fires _before_ the VC has been recreated (say after login or logout) then this signal doesn't emit. However, we also don't want it to always emit twice - once with the cached value, and again with the updated value. The debounce attempts to solve this by "waiting" for 1 second for the updated config notification: if the notification arrives, then only the updated config is taken to determine the `showEditorialHeader` value. If we miss the notification, then the signal will emit with the cached value.